pkg/detectors/azurefunctionkey/azurefunctionkey.go

```diff
@@ -0,0 +1,89 @@
+package azurefunctionkey
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	defaultClient = common.SaneHttpClient()
+	// Make sure that your group is surrounded in boundary characters such as below to reduce false positives.
+	keyPat      = regexp.MustCompile(`\b[a-zA-Z0-9_-]{54,56}==`)
```

Comment:
It looks like this might be base64, so the ending might not always be `==`. Can you check a few more keys to confirm? Also, we should probably have a PrefixRegex here since it is not a very unique pattern.

Reply:
Also is this for all access key types? Is there any way to tell the difference?

https://learn.microsoft.com/en-us/azure/azure-functions/security-concepts?tabs=v4#function-access-keys

Reply:
I just updated the guide for this https://github.com/trufflesecurity/how-to-rotate/pull/16